packages/subproviders/src/subproviders/mnemonic_wallet.ts

```diff
@@ -0,0 +1,147 @@
+const DEFAULT_NUM_ADDRESSES_TO_FETCH = 10;
+const DEFAULT_ADDRESS_SEARCH_LIMIT = 1000;
+const INITIAL_KEY_DERIVATION_INDEX = 0;
+
+/**
+ * This class implements the [web3-provider-engine](https://github.com/MetaMask/provider-engine) subprovider interface.
+ * This subprovider intercepts all account related RPC requests (e.g message/transaction signing, etc...) and handles
+ * all requests with accounts derived from the supplied mnemonic.
+ */
+export class MnemonicWalletSubprovider extends BaseWalletSubprovider {
+    private _addressSearchLimit: number;
+    private _baseDerivationPath: string;
+    private _derivedKeyInfo: DerivedHDKeyInfo;
+    private _mnemonic: string;
+
+    /**
+     * Instantiates a MnemonicWalletSubprovider. Defaults to baseDerivationPath set to `44'/60'/0'/0`.
+     * This is the default in TestRPC/Ganache, it can be overridden if desired.
+     * @param config Configuration for the mnemonic wallet, must contain the mnemonic
+     * @return MnemonicWalletSubprovider instance
+     */
+    constructor(config: MnemonicWalletSubproviderConfigs) {
+        const mnemonic = config.mnemonic;
+        assert.isString('mnemonic', mnemonic);
+        const baseDerivationPath = config.baseDerivationPath || DEFAULT_BASE_DERIVATION_PATH;
+        assert.isString('baseDerivationPath', baseDerivationPath);
+        const addressSearchLimit = config.addressSearchLimit || DEFAULT_ADDRESS_SEARCH_LIMIT;
+        assert.isNumber('addressSearchLimit', addressSearchLimit);
+        super();
+
+        this._mnemonic = mnemonic;
+        this._baseDerivationPath = baseDerivationPath;
+        this._addressSearchLimit = addressSearchLimit;
+        this._derivedKeyInfo = this._initialDerivedKeyInfo(this._baseDerivationPath);
+    }
+    /**
+     * Retrieve the set derivation path
+     * @returns derivation path
+     */
+    public getPath(): string {
+        return this._baseDerivationPath;
+    }
+    /**
+     * Set a desired derivation path when computing the available user addresses
+     * @param baseDerivationPath The desired derivation path (e.g `44'/60'/0'`)
+     */
+    public setPath(baseDerivationPath: string) {
+        this._baseDerivationPath = baseDerivationPath;
+        this._derivedKeyInfo = this._initialDerivedKeyInfo(this._baseDerivationPath);
+    }
+    /**
+     * Retrieve the accounts associated with the mnemonic.
+     * This method is implicitly called when issuing a `eth_accounts` JSON RPC request
+     * via your providerEngine instance.
+     * @param numberOfAccounts Number of accounts to retrieve (default: 10)
+     * @return An array of accounts
+     */
+    public async getAccountsAsync(numberOfAccounts: number = DEFAULT_NUM_ADDRESSES_TO_FETCH): Promise<string[]> {
+        const derivedKeys = walletUtils.calculateDerivedHDKeyInfos(this._derivedKeyInfo, numberOfAccounts);
+        const accounts = _.map(derivedKeys, k => k.address);
+        return accounts;
+    }
+
```

Comment:
Remove spaces. Why do these get added in? Thought we had a TSLint rule for these.

Reply:
Let's add an Airtable task for creating one.